### PR TITLE
Simplify sorted set collectors

### PR DESCRIPTION
**Pull request by idegtiarenko**
Replace custom collector classes with a generic `Collector.of(..)` call.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/common/util/set/Sets.java [hunk 1]
```diff
@@ -137,7 +132,7 @@ public static <T> SortedSet<T> unmodifiableSortedDifference(final Set<T> left, f
      * @return a sorted set
      */
     public static <T> Collector<T, SortedSet<T>, SortedSet<T>> toSortedSet() {
-        return new SortedSetCollector<>();
+        return Collector.of(TreeSet::new, SortedSet::add, Sets::addAllMutable);
```
Comment (idegtiarenko) [inline server/src/main/java/org/elasticsearch/common/util/set/Sets.java:135]: This adds `Characteristics.IDENTITY_FINISH` characteristic internally.

**Review by pxsalehi — APPROVED** [review]
LGTM!

File: server/src/main/java/org/elasticsearch/common/util/set/Sets.java [hunk 2]
```diff
@@ -148,63 +143,12 @@ public static <T> Collector<T, SortedSet<T>, SortedSet<T>> toSortedSet() {
...
-        public Function<SortedSet<T>, SortedSet<T>> finisher() {
-            return Collections::unmodifiableSortedSet;
-        }
-
-        @Override
-        public Set<Characteristics> characteristics() {
-            return Collections.emptySet();
-        }
-
+    private static <T, S extends Set<T>> S addAllMutable(S a, S b) {
```
Comment (arteam) [inline server/src/main/java/org/elasticsearch/common/util/set/Sets.java:149]: I'm late for the review, but the combiner function should generate a new set combined from `a` and `b`. It exists for the case if a stream is parallel and the end result needs to be merged from sub-results produced by different threads.

Comment (idegtiarenko) [reply server/src/main/java/org/elasticsearch/common/util/set/Sets.java:149]: I replicated this behavior from the previous implementation, possibly this collector was never used in a parallel stream?

I checked a javadoc for `java.util.stream.Collector#combiner` and it states:
```
A function that accepts two partial results and merges them. The combiner function may fold state from one argument into the other and return that, or may return a new result container.
```
Is there a clarification somewhere else for a concurrent case?

**Pull request merged**